### DISCUSSION TITLE
Spring Cloud AWS SQS Integration

### DISCUSSION THREAD

**Discussion by tomazfernandes** [spring-projects/spring-kafka#2320] [General]
Hi, 

I'd like to properly share with you that I've been working on designing and implementing the `SQS` integration for the `Spring Cloud AWS` project. That's a complete rewrite of the module, so it's being written from scratch. Quite a challenge! 😄 

I've been mainly using `Spring Kafka` as a starting point for the design, while also trying to simplify some things where possible since that's a large umbrella project with a small team of volunteers doing maintenance.

I'm not sure if and how I should give proper credit to the creators of this project - while I'm not copying classes, the overall assembly-phase design is very similar. Please let me know if you have any suggestions.

So first I'd like to thank you both once again for everything I've been learning from you for the past year and a half - hopefully I'm doing a better job design-wise there than what I did with the `RetryableTopic` feature 😄 

I'd also like to invite you to take a look at what we have so far - I've recently reviewed the entire design and, while there's still a lot to be done, I'm confident enough with what we have. I won't tag you there because it's not something specifically related with this project and I know you're busy with your many duties, but if you do have the time and interest, that would be very welcome.

After all, if any good comes out of this integration, that's the result of the brilliant work you've been doing here, and the time you've been investing in assisting me with my contributions to this project.

Also, it is my understanding that at some point in the future this might be used by the `Spring Integration` project, so perhaps you'd like to have the opportunity to anticipate any potential issues.

This is the open draft: https://github.com/awspring/spring-cloud-aws/pull/374
There are also a few open issues, in case you'd like to take a look: https://github.com/awspring/spring-cloud-aws/issues

Thank you both very much!

**Comment by artembilan** [comment]
There is already Spring Integration AWS project, which is fully based based on the mentioned Spring Cloud AWS.
Well, at least in that SQS part: https://github.com/spring-projects/spring-integration-aws/#amazon-simple-queue-service-sqs.

I also have plans to look into migrating to AWS SDK v2: https://github.com/spring-projects/spring-integration-aws/issues/155.

And yes: we have a Kinesis channel adapters support over there as well.

If this is a proposal of contribution, then you are very welcome! 😄

**Reply by tomazfernandes** [reply]
> There is already Spring Integration AWS project, which is fully based based on the mentioned Spring Cloud AWS. Well, at least in that SQS part: https://github.com/spring-projects/spring-integration-aws/#amazon-simple-queue-service-sqs.

Sure, one of the requisites I got was that in the end we should have an `SQS` integration that could later be used by `Spring Integration`, so that's why I brought this up.

> If this is a proposal of contribution, then you are very welcome! 😄

Yeah, maybe in the future!
